apps/prairielearn/src/ee/lib/aiQuestionGeneration.ts

```diff
@@ -87,14 +144,98 @@ Keep in mind you are not just generating an example; you are generating an actua
       user: openAiUserFromAuthn(authnUserId),
     });
 
-    const completionText = completion.choices[0].message.content;
+    extractFromCompletion(completion, job);
+    job.data['prompt'] = prompt;
+    job.data['generation'] = completion.choices[0].message.content;
+    job.data['context'] = context;
```

Comment:
I might consider including the entire `completion` in `job.data`, both for debugging and so that we can store + retrieve usage information in a structured way.

Reply:
done in e0331beaf9d587990dcda087d800feedc786fc87.